src/listconf.c

```diff
@@ -345,6 +349,18 @@ static void listconf_list_build_info(void)
 	printf("Terminal locale string: %s\n", john_terminal_locale);
 	printf("Parsed terminal locale: %s\n", cp_id2name(options.terminal_enc));
 
+#if defined (__CYGWIN__)
+	{
+		struct utsname buffer;
+
+		errno = 0;
+		if (uname(&buffer) != 0)
+			printf("Cygwin version detection error");
```

Comment:
Add `\n` or use `puts`.

Also, why reset `errno` if we don't check it later? Even if we did use `errno`, per POSIX it should be safe to do so after `uname` returns failure - but success is indicated by `a non-negative value`, not just 0. Perhaps Cygwin has its own documentation making this more specific.

POSIX:

```
RETURN VALUE
       Upon  successful completion, a non-negative value shall be returned.
       Otherwise, -1 shall be returned and errno set to indicate the error.
```